### PR TITLE
Feat: Add support to specify nginx ingress controller ssl protocols

### PR DESCRIPTION
**Pull request by nightfury1204**
### What is the feature/fix?

https://app.asana.com/0/1203637156732418/1204034344868223/f

### Does it has a breaking change?

no.
setting invalid values will make ssl handshake failure

### How to use/test it?

- deploy rack of this version
- set the param: `convox rack params set ssl_protocols="TLSv1.1 TLSv1.2 TLSv1.3" ssl_ciphers="ALL"`

### Checklist
- [ ] New coverage tests
- [ ] Unit tests passing
- [ ] E2E tests passing
- [ ] E2E downgrade/update test passing
- [ ] Documentation updated
- [ ] No warnings or errors on Deepsource/Codecov

### CODE REVIEW

**Comment by yuriadams** [comment]
I believe you need to change your base branch to `staging`, @nightfury1204

**Comment by yuriadams** [comment]
I believe you need to change your base branch to `staging`, @nightfury1204

**Comment by yuriadams** [comment]
@nightfury1204, didn't you need to change the ciphers as well when you put a old TLS version? I made a quick test a few days ago and it broke.

**Comment by nightfury1204** [comment]
> @nightfury1204, didn't you need to change the ciphers as well when you put a old TLS version? I made a quick test a few days ago and it broke.
@yuriadams 
i checked for TLSv1.2 and TLSv1.3. looks ok.
But we shouldn't support deprecated tls(it's problematic)
```
In October 2018, Apple, Google, Microsoft, and Mozilla jointly announced they would deprecate TLS 1.0 and 1.1 in March 2020. TLS 1.0 and 1.1 were formally deprecated in RFC 8996 in March 2021.
```
Let me know your thoughts

**Comment by yuriadams** [comment]
hey, @nightfury1204. @ntner can confirm but the customer was trying to use an old tls version.

File: docs/installation/production-rack/aws.md [hunk 1]
```diff
@@ -63,6 +63,7 @@ The following environment variables are required:
 | **proxy_protocol**       | **false**               | Enable proxy protocol. With this parameter set, the client source ip will be available in the request header `x-forwarded-for` key. **Requires 5 - 10 minutes downtime**.          |
 | **region**               | **us-east-1**          | AWS Region                                                                                                     |
 | **syslog**               |                        | Forward logs to a syslog endpoint (e.g. **tcp+tls://example.org:1234**)                                        |
+| **ssl_protocols**        |                        | SSL protocols to use for (nginx)[https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_ssl_protocols] (e.g. **TLSv1.2 TLSv1.3**)|
```
Comment (Twsouza) [inline docs/installation/production-rack/aws.md:66]:
```suggestion
| **ssl_protocols**        |                        | SSL protocols to use for (nginx)[https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_ssl_protocols] (e.g. **TLSv1.2 TLSv1.3**)|. They must be separated by spaces.
```

**Review by Twsouza — CHANGES_REQUESTED** [review]
looks good, just a minor addition.

**Comment by nightfury1204** [comment]
@yuriadams 

```
$ curl --tls-max 1.1 --ciphers DEFAULT@SECLEVEL=0 -v https://web.nodejs.5d32949470eb534a.convox.cloud -k
*   Trying 18.214.101.78:443...
* Connected to web.nodejs.5d32949470eb534a.convox.cloud (18.214.101.78) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* Cipher selection: DEFAULT@SECLEVEL=0
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.1 (OUT), TLS handshake, Client hello (1):
* TLSv1.1 (IN), TLS header, Certificate Status (22):
* TLSv1.1 (IN), TLS handshake, Server hello (2):
* TLSv1.1 (IN), TLS header, Certificate Status (22):
* TLSv1.1 (IN), TLS handshake, Certificate (11):
* TLSv1.1 (IN), TLS header, Certificate Status (22):
* TLSv1.1 (IN), TLS handshake, Server key exchange (12):
* TLSv1.1 (IN), TLS header, Certificate Status (22):
* TLSv1.1 (IN), TLS handshake, Server finished (14):
* TLSv1.1 (OUT), TLS header, Certificate Status (22):
* TLSv1.1 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.1 (OUT), TLS header, Finished (20):
* TLSv1.1 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.1 (OUT), TLS header, Certificate Status (22):
* TLSv1.1 (OUT), TLS handshake, Finished (20):
* TLSv1.1 (IN), TLS header, Finished (20):
* TLSv1.1 (IN), TLS header, Certificate Status (22):
* TLSv1.1 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.1 / ECDHE-RSA-AES256-SHA
* ALPN, server accepted to use h2
* Server certificate:
*  subject: O=Acme Co; CN=Kubernetes Ingress Controller Fake Certificate
*  start date: Mar 28 14:10:57 2023 GMT
*  expire date: Mar 27 14:10:57 2024 GMT
*  issuer: O=Acme Co; CN=Kubernetes Ingress Controller Fake Certificate
*  SSL certificate verify result: self-signed certificate (18), continuing anyway.
* Using HTTP2, server supports multiplexing
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* TLSv1.1 (OUT), TLS header, Supplemental data (23):
* TLSv1.1 (OUT), TLS header, Supplemental data (23):
* TLSv1.1 (OUT), TLS header, Supplemental data (23):
* Using Stream ID: 1 (easy handle 0x55c17d3e2150)
* TLSv1.1 (OUT), TLS header, Supplemental data (23):
> GET / HTTP/2
> Host: web.nodejs.5d32949470eb534a.convox.cloud
> user-agent: curl/7.81.0
> accept: */*
> 
* TLSv1.1 (IN), TLS header, Supplemental data (23):
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
* TLSv1.1 (OUT), TLS header, Supplemental data (23):
* TLSv1.1 (IN), TLS header, Supplemental data (23):
* TLSv1.1 (IN), TLS header, Supplemental data (23):
< HTTP/2 200 
< date: Tue, 28 Mar 2023 15:11:34 GMT
< content-type: text/plain
< content-length: 92
< strict-transport-security: max-age=15724800; includeSubDomains
< 
Hello World! test2
I'm: nodejs
Build:BKBWOIVRKNV
Release: RCTTYWENKFI
* TLSv1.1 (IN), TLS header, Supplemental data (23):
* Connection #0 to host web.nodejs.5d32949470eb534a.convox.cloud left intact
running on: v3-rack118% 
```

**Pull request merged**